components/exercises/chapter-4-oscar/DeepComparison/DeepComparison.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { isDeepEqual } from './isDeepEqual';
+
+/**
+ * Represents the properties of a week item.
+ */
+export type ShopItem = {
+  name: string;
+  description: string;
+  price: number;
+};
+
+/**
+ * This represents an array of arrays of WeekItem objects, essentially creating a two-dimensional array of WeekItem objects. Each element of the outer array is an array of WeekItem objects.
+ */
+type GroceryList = ShopItem[][];
+
+const newElement: ShopItem = {
+  name: '🍅 Tomatoes',
+  description: '1kg Fresh tomatoes',
+  price: 0.5,
+};
+
+export const lastWeekShoppingList: ShopItem[] = [
+  {
+    name: '🥚 Eggs',
+    description: '6x Large class A free range eggs',
+    price: 2.5,
+  },
+  {
+    name: '🐓 Chicken',
+    description: '650gr Fresh Class A skinless chicken breast fillet portions.',
+    price: 4.7,
+  },
+  {
+    name: '🍚 Basmati rice',
+    description:
+      '1kg Aromatic & fluffy A long, slender grain grown in the foothills of the Himalayas',
+    price: 2.5,
+  },
+  {
+    name: '🥔 Potatoes',
+    description: '1kg White potatoes',
+    price: 1.5,
+  },
+];
+
+export const currentWeekShoppingList: ShopItem[] = [
+  {
+    name: '🥚 Eggs',
+    description: '6x Large class A free range eggs',
+    price: 2.5,
+  },
+  {
+    name: '🐓 Chicken',
+    description: '650gr Fresh Class A skinless chicken breast fillet portions.',
+    price: 4.7,
+  },
+  {
+    name: '🍚 Basmati rice',
+    description:
+      '1kg Aromatic & fluffy A long, slender grain grown in the foothills of the Himalayas',
+    price: 2.5,
+  },
+  {
+    name: '🥔 Potatoes',
+    description: '1kg White potatoes',
+    price: 1.5,
+  },
+];
+
+/**
+ * Renders a component that compares two grocery lists and allows adding/removing items and checks for deep equality of both grocery lists.
```

Comment:
* Renders a component that compares two grocery lists, allows adding/removing items and checks for deep equality of both grocery lists.